src/test/java/tests/junit/Tests.java

```diff
@@ -0,0 +1,195 @@
+        Assertions.assertTrue(realItem.toString().equals(String.format(
+                "Class: %s; Name: %s; Price: %s; Weight: %s",
+                realItem.getClass(),
+                realItem.getName(),
+                realItem.getPrice(),
+                realItem.getWeight())
+        ));
+    }
+
+    @Test
+    @DisplayName("Check VirtualItem")
+    public void checkVirtualItem(){
+        VirtualItem virtualItem = new VirtualItem();
+        virtualItem.setName("windows");
+        virtualItem.setPrice(125.00);
+        virtualItem.setSizeOnDisk(2699.00);
+        Assertions.assertTrue(virtualItem.toString().equals(String.format(
+                "Class: %s; Name: %s; Price: %s; Size on disk: %s",
+                virtualItem.getClass(),
+                virtualItem.getName(),
+                virtualItem.getPrice(),
+                virtualItem.getSizeOnDisk()
+        )));
+    }
+
+    @ParameterizedTest
+    @ValueSource(ints = {5})
+    @DisplayName("Check addVirtualItem method")
+    public void checkAddVirtualItem(int argument){
+        double originalPrice = carts.get(argument).getTotalPrice();
+        VirtualItem virtualItem = new VirtualItem();
+        virtualItem.setName("Home Design");
+        virtualItem.setPrice(98.00);
+        virtualItem.setSizeOnDisk(190.00);
+        Cart cart = carts.get(argument);
+        cart.addVirtualItem(virtualItem);
+
+        Assertions.assertTrue(cart.getTotalPrice() == originalPrice + virtualItem.getPrice()*1.2);
+    }
+
+    @ParameterizedTest
+    @ValueSource(ints = {5})
+    @DisplayName("Check deleteVirtualItem method")
+    public void checkDeleteVirtualItem(int argument){
+        double totalPrice = 0.d;
+        Cart cart = carts.get(argument);
+        VirtualItem virtualItem = new VirtualItem();
+        virtualItem.setName("Home Design");
+        virtualItem.setPrice(98.00);
+        virtualItem.setSizeOnDisk(190.00);
+        cart.addVirtualItem(virtualItem);
+        totalPrice=cart.getTotalPrice();
+        cart.deleteVirtualItem(virtualItem);
+        Assertions.assertTrue(totalPrice > cart.getTotalPrice());
+    }
+
+    @ParameterizedTest
+    @ValueSource(ints = {5})
+    @DisplayName("Check addRealItem method")
+    public void checkAddRealItem(int argument){
+        double originalPrice = carts.get(argument).getTotalPrice();
+        RealItem realItem = new RealItem();
+        realItem.setName("Audi");
+        realItem.setPrice(25698.00);
+        realItem.setWeight(1900.03);
+        Cart cart = carts.get(argument);
+        cart.addRealItem(realItem);
+
+        Assertions.assertTrue(cart.getTotalPrice() == originalPrice + realItem.getPrice()*1.2);
+    }
+
+    @ParameterizedTest
+    @ValueSource(ints = {5})
+    @DisplayName("Check deleteRealItem method")
+    public void checkDeleteRealItem(int argument){
+        double totalPrice = 0.d;
+        Cart cart = carts.get(argument);
+        RealItem realItem = new RealItem();
```

Comment:
please think how to reduce code duplication in this test and several above

Reply:
fixed